src/enable_variable_ops/ngraph_replace_optimizers.cc

```diff
@@ -0,0 +1,208 @@
+/*******************************************************************************
+ * Copyright 2019-2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+#include "ngraph_replace_optimizers.h"
+#include "ngraph_api.h"
+#include "ngraph_capture_variables.h"
+#include "ngraph_utils.h"
+#include "tensorflow/core/graph/graph.h"
+#include "tensorflow/core/graph/node_builder.h"
+
+using namespace std;
+namespace ng = ngraph;
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+Status ReplaceOptimizers(Graph* graph, int graph_id) {
+  // Go over the nodes and replace variable modifiers with the computation graph
+  // Add Assign Op in their place
+
+  for (auto node : graph->op_nodes()) {
+    if (node->type_string() == "NGraphAssignSub" ||
+        node->type_string() == "NGraphAssignAdd") {
+      NodeBuilder::NodeOut input_ref;
+      NodeBuilder::NodeOut input_val;
+
+      DataType dtype;
+      TF_RETURN_IF_ERROR(GetNodeAttr(node->attrs(), "T", &dtype));
+
+      for (auto edge : node->in_edges()) {
+        if (edge == NULL) {
+          continue;
+        }
+        if (edge->dst()->IsOp() && !edge->IsControlEdge() &&
+            IsRefType(edge->dst()->input_type(edge->dst_input()))) {
+          input_ref = NodeBuilder::NodeOut(edge->src(), edge->src_output());
+        } else {
+          input_val = NodeBuilder::NodeOut(edge->src(), edge->src_output());
+        }
+      }
+
+      string op_type =
+          (node->type_string() == "NGraphAssignSub") ? "Sub" : "Add";
+      string op_name_suffix = "_" + op_type;
+      Node* compute_op;
+      string new_name_sub = node->name() + op_name_suffix;
```

Comment:
In kanvi's PR she uses `NewName` to get a guaranteed unique name. Maybe we should use that API to generate new names

Reply:
the current op name is unique, so adding the suffix would still keep it unique